x-pack/plugins/observability_solution/infra/server/routes/infra/lib/helpers/query.ts

```diff
@@ -5,41 +5,34 @@
  * 2.0.
  */
 
-import { estypes } from '@elastic/elasticsearch';
 import { findInventoryModel } from '@kbn/metrics-data-access-plugin/common';
-import { termsQuery, rangeQuery } from '@kbn/observability-plugin/server';
+import { termQuery, termsQuery } from '@kbn/observability-plugin/server';
 import {
-  GetInfraMetricsRequestBodyPayload,
-  InfraAssetMetricType,
-} from '../../../../../common/http_api/infra';
-import { BUCKET_KEY } from '../constants';
+  EVENT_MODULE,
+  METRICSET_MODULE,
+  METRICSET_NAME,
+  SYSTEM_INTEGRATION,
+} from '../../../../../common/constants';
+import { InfraAssetMetricType } from '../../../../../common/http_api/infra';
 
-export const createFilters = ({
-  params,
-  extraFilter,
-  hostNamesShortList = [],
-}: {
-  params: GetInfraMetricsRequestBodyPayload;
-  hostNamesShortList?: string[];
-  extraFilter?: estypes.QueryDslQueryContainer;
-}) => {
-  const extrafilterClause = extraFilter?.bool?.filter;
-
-  const extraFilterList = !!extrafilterClause
-    ? Array.isArray(extrafilterClause)
-      ? extrafilterClause
-      : [extrafilterClause]
-    : [];
+export const getFilterByIntegration = (integration: typeof SYSTEM_INTEGRATION) => {
+  return {
+    bool: {
+      should: [
+        ...termQuery(EVENT_MODULE, integration),
+        ...termQuery(METRICSET_MODULE, integration),
+      ],
+      minimum_should_match: 1,
+    },
+  };
+};
 
+export const getValidDocumentsFilter = () => {
   return [
-    ...extraFilterList,
-    ...termsQuery(BUCKET_KEY, ...hostNamesShortList),
-    ...rangeQuery(new Date(params.range.from).getTime(), new Date(params.range.to).getTime()),
-    {
-      exists: {
-        field: BUCKET_KEY,
-      },
-    },
+    // system module
+    getFilterByIntegration('system'),
+    // apm docs
+    ...termsQuery(METRICSET_NAME, 'transaction'),
```

Comment:
I'm not sure if this is the right filter

Reply:
@cauemarcondes @MiriamAparicio do you know if filtering by  `metricset.name: 'transaction'` is correct if I want to get metrics for hosts from APM?

Reply:
as we talked, you also need to inform the interval. https://www.elastic.co/guide/en/observability/current/apm-data-model-metrics.html#_service_summary_metrics